### PR TITLE
[tests-only] Fixed regex pattern used in download content matching

### LINKED ISSUE #38300: Test step for downloaded content is not strict enough
Example: this test scenario passes:
```
make test-acceptance-api BEHAT_FEATURE=tests/acceptance/features/apiProvisioning-v1/disableUser.feature:179
```

Remove the spaces from the expected content in the last step:
```
  Scenario: getting shares shared by disabled user
    Given the administrator has set the default folder for received shares to "Shares"
    And auto-accept shares has been disabled
    And user "Alice" has been created with default attributes and skeleton files
    And user "Brian" has been created with default attributes and without skeleton files
    And user "Alice" has shared file "/textfile0.txt" with user "Brian"
    And user "Brian" has accepted share "/textfile0.txt" offered by user "Alice"
    When the administrator disables user "Alice" using the provisioning API
    Then as "Brian" file "/Shares/textfile0.txt" should exist
    And the content of file "/Shares/textfile0.txt" for user "Brian" should be "ownCloudtesttextfile0"
```

The test still passes. But it should fail, because the actual content is not what the test expects.

https://github.com/owncloud/core/pull/38247/files#diff-730f2be358a1fc6a85ee5100526fb1bbcfc003547649bf9133ef13f4674a0dadR886 added some regex logic to the content comparison. But that seems to result in white-space differences being ignored.

Investigate. Maybe there are some test steps where we want to ignore some differences (maybe end-of-line characters, or space at the beginning or end or?) - but mostly we should be looking for an exact match.

### PR DESCRIPTION
**Pull request by Talank**
## Related Issue
- Fixes #38300 

## How Has This Been Tested?
CI

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by phil-davis** [comment]
See my comment https://github.com/owncloud/core/issues/38300#issuecomment-764394701

IMO it will be easier and clearer code if we remove the 3 lines of special `preg_replace` stuff from `downloadedContentShouldBe` and make a separate step that does a not-exact "special" match of the content. Maybe the "special" code is only needed for the new step "the downloaded content for multipart byterange should be" `theDownloadedContentForMultipartByterangeShouldBe` and so the special `preg_replace` code can be put in there.

**Comment by saw-jan** [comment]
@Talank also if it is to remove extra space before and after the text content,  `trim()` will do the job well.

**Comment by Talank** [comment]
> @Talank also if it is to remove extra space before and after the text content, `trim()` will do the job well.

it's more about preserving the spaces between words, The trimming is done perfectly by the special `preg_replace` stuff

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 1]
```diff
@@ -903,7 +900,16 @@ public function downloadedContentShouldBe($content) {
 	public function theDownloadedContentForMultipartByterangeShouldBe($statusCode, PyStringNode $content) {
 		$actualStatusCode = $this->response->getStatusCode();
 		if ($actualStatusCode === $statusCode) {
-			$this->downloadedContentShouldBe($content->getRaw());
+			$actualContent = (string) $this->response->setBody();
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebDav.php:903]:
```suggestion
			$actualContent = (string) $this->response->getBody();
```

This "looks wrong"

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 2]
```diff
@@ -903,7 +900,16 @@ public function downloadedContentShouldBe($content) {
...
+			$pattern = ["/--\w*/", "/\s*/m"];
+			$actualContent = \preg_replace($pattern, "", $actualContent);
+			$content = \preg_replace("/\s*/m", '', $content->getRaw());
+			Assert::assertEquals(
+				$content,
+				$actualContent,
+				"The downloaded content was expected to be '$content', but actually is '$actualContent'. HTTP status was $actualStatus"
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebDav.php:910]:
```suggestion
				"The downloaded content was expected to be '$content', but actually is '$actualContent'. HTTP status was $actualStatusCode"
```

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 3]
```diff
@@ -903,7 +900,16 @@ public function downloadedContentShouldBe($content) {
...
+			);
+			$this->downloadedContentShouldBe($content);
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebDav.php:912]: Why do we call this? I thought that the content was already compared in the `Assert::assertEquals` call above.

**Review by phil-davis — DISMISSED** [review]

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/core/28384/57/13
Some scenarios are failing, for example:
```
  Scenario: getting shares shared by disabled user                                                         # /drone/src/tests/acceptance/features/apiProvisioning-v1/disableUser.feature:179
    Given the administrator has set the default folder for received shares to "Shares"                     # OccContext::theAdministratorHasSetTheDefaultFolderForReceivedSharesTo()
    And auto-accept shares has been disabled                                                               # FeatureContext::autoAcceptSharesHasBeenDisabled()
    And user "Alice" has been created with default attributes and skeleton files                           # FeatureContext::userHasBeenCreatedWithDefaultAttributes()
    And user "Brian" has been created with default attributes and without skeleton files                   # FeatureContext::userHasBeenCreatedWithDefaultAttributesAndWithoutSkeletonFiles()
    And user "Alice" has shared file "/textfile0.txt" with user "Brian"                                    # FeatureContext::userHasSharedFileWithUserUsingTheSharingApi()
    And user "Brian" has accepted share "/textfile0.txt" offered by user "Alice"                           # FeatureContext::userHasReactedToShareOfferedBy()
    When the administrator disables user "Alice" using the provisioning API                                # FeatureContext::adminDisablesUserUsingTheProvisioningApi()
    Then as "Brian" file "/Shares/textfile0.txt" should exist                                              # FeatureContext::asFileOrFolderShouldExist()
    And the content of file "/Shares/textfile0.txt" for user "Brian" should be "ownCloud test text file 0" # FeatureContext::contentOfFileForUserShouldBe()
      The downloaded content was expected to be 'ownCloud test text file 0', but actually is 'ownCloud test text file 0
      '. HTTP status was 200
      Failed asserting that two strings are equal.
      --- Expected
      +++ Actual
      @@ @@
      -'ownCloud test text file 0'
      +'ownCloud test text file 0\n
      +'
```

There is a step `contentOfFileForUserShouldBePlusEndOfLine` that should work for those scenarios - `textfile0.txt` has an end-of-line character.

Also the similar fails in https://drone.owncloud.com/owncloud/core/28384/58/13

**Comment by Talank** [comment]
@phil-davis I think the idea of completely removing the `preg_replace()` from `downloadedContentShouldBe` is not that accurate. I will add a `preg_replace()` to filter the `\n`

**Review by phil-davis — CHANGES_REQUESTED** [review]
We have a step already to specifically check for the end-of-line character. We want the tests to check the exact file content (we do not want the test to still pass if the downloaded content has an extra `\n` or is missing the `n`.

We can just use the step.

**Review by phil-davis — APPROVED** [review]

**Pull request merged**